### PR TITLE
URI-escape image filenames

### PR DESCRIPTION
**Pull request by eric-wieser**
Without this change, local images with `#` in their name result in incorrect URLs in HTML builds.

There is already a similar call to `urllib.parse.quote` for file downloads, added in #9670, suggesting this is a sensible approach.

This also adds a hack to make things work for `#` specifically in LaTeX, but more work is likely needed for other weird character there.

### Feature or Bugfix

- Bugfix

### CODE REVIEW

**Review by tk0miya — CHANGES_REQUESTED** [review]
Looks good. Could you add a testcase for this, please? Then I'll merge this soon.

**Comment by eric-wieser** [comment]
#9670 didn't have a test-case, so I was hoping I could get away without one here too...

I might have time for a test-case in a week or so, we'll see.

**Comment by tk0miya** [comment]
It seems #9670 contains a test-case :-)

**Comment by tk0miya** [comment]
Hmm... epubcheck and LaTeX build are failed when we use `#` for the filename. So escaping is not a good way to handle such files. We need to rename them on copying to the generated document.

**Comment by eric-wieser** [comment]
It sounds like the latex code needs escaping too

**Comment by eric-wieser** [comment]
Thanks for fixing the broken tests!

**Comment by eric-wieser** [comment]
Is there any way to see the latex error message from CI?

**Comment by eric-wieser** [comment]
I think the CI failure is actually a bug in `epubcheck`

File: sphinx/builders/_epub_base.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from os import path
 from typing import Any, Dict, List, NamedTuple, Optional, Set, Tuple
 from zipfile import ZIP_DEFLATED, ZIP_STORED, ZipFile
+from urllib.parse import quote
```
Comment (AA-Turner) [inline sphinx/builders/_epub_base.py]:
```suggestion
from urllib.parse import quote
from zipfile import ZIP_DEFLATED, ZIP_STORED, ZipFile
```

**Comment by AA-Turner** [comment]
@eric-wieser it looks like the test failures are pretty simple to solve--are you up for having a look, it would be nice to get this in to 5.3.

A

File: CHANGES [hunk 2]
```diff
@@ -20,6 +20,7 @@ Features added
   ``:option:`--module[=foobar]``` or ``:option:`--module foobar```.
   Patch by Martin Liska.
 * #10881: autosectionlabel: Record the generated section label to the debug log.
+# 10268: Correctly URI-escape image filenames.
```
Comment (AA-Turner) [inline CHANGES]:
```suggestion
* #10268: Correctly URI-escape image filenames.
```

**Comment by AA-Turner** [comment]
Thanks @eric-wieser!

A

**Pull request merged**